src/trace/trace.c

```diff
@@ -109,31 +109,23 @@ static inline bool trace_filter_pass(uint32_t lvl,
 	return lvl <= ctx->level;
 }
 
-void trace_log(bool send_atomic, const void *log_entry,
-	       const struct tr_ctx *ctx, uint32_t lvl, uint32_t id_1,
-	       uint32_t id_2, int arg_count, ...)
+static inline void vatrace_log(bool send_atomic, uint32_t log_entry, const struct tr_ctx *ctx,
+			       uint32_t lvl, uint32_t id_1, uint32_t id_2, int arg_count,
+			       va_list vargs)
```

Comment:
I thought about changing `id_1` and `id_2` to `pipe_id` and `comp_id`, I didn't do that yet, but maybe it's good PR for that, because those names are duplicated a few times here.